cmd/clusterctl/client/cluster/mover_test.go

```diff
@@ -788,3 +788,70 @@ func Test_objectsMoverService_checkTargetProviders(t *testing.T) {
 		})
 	}
 }
+
+func Test_objectMoverService_ensureNamespace(t *testing.T) {
+	type args struct {
+		toProxy   Proxy
+		namespace string
+	}
+
+	namespace1 := &corev1.Namespace{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Namespace",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "namespace-1",
+		},
+	}
+
+	tests := []struct {
```

Comment:
Do we want to take this chance to add test cases for when it does return an error?

Reply:
That is what I initially had in mind

Reply:
TBH, simulating errors is not simple (RBAC failures or FakeClient failures), so I'm ok with the current coverage (for sure we are in better shape than before)